lib/relaton/index/type.rb

```diff
@@ -38,7 +39,7 @@ def actual?(**args)
       #
       # Add or update index item
       #
-      # @param [String] id document ID
+      # @param [Pubid::Core::Identifier::Base] id document ID
```

Comment:
If PubId is added to index how will it be serialized when index is saved into YAML file?

Reply:
@andrew2net you can see it here https://github.com/metanorma/pubid-core/commit/5ac4fff095783867ab4b795d55f84420e9481811

Reply:
@mico so the YAML will be like:
```ruby
> id = Pubid::Ccsds::Identifier.parse 'CCSDS 230.2-G-1'
 => #<Pubid::Ccsds::Identifier::Base:0x0000000100add350 @book_color="G", @edition="1", @number="230", @part=2, @publisher="CCSDS", @retired=false>
 :042 > idx = [{id: id, file: 'file.yam'}]

 > idx = [{id: id, file: 'file.yam'}]
 => [{:id=>#<Pubid::Ccsds::Identifier::Base:0x0000000100add350 @book_color="G", @edition="1", @number="230", @part=2, @publisher="CCSDS", @retired=false>, :file=>"file.yam"}

> puts idx.to_yaml
---
- :id: !ruby/object:Pubid::Ccsds::Identifier::Base
    publisher: CCSDS
    number: '230'
    book_color: G
    part: 2
    retired: false
    edition: '1'
  :file: file.yam
```
I thought we decided not to save class names in YAML file.

Reply:
@mico I have installed pubid-core v1.12.9, should I downgrade to v1.12.8?
Yes, dependency shouldn't allow using wrong version.